src/plugins/kdb/ldap/libkdb_ldap/ldap_principal2.c

```diff
@@ -1375,64 +1493,221 @@ krb5_read_tkt_policy(krb5_context context, krb5_ldap_context *ldap_context,
     return st;
 }
 
-krb5_error_code
-krb5_decode_krbsecretkey(krb5_context context, krb5_db_entry *entries,
-                         struct berval **bvalues,
-                         krb5_tl_data *userinfo_tl_data, krb5_kvno *mkvno)
+static void
+free_ldap_seqof_key_data(ldap_seqof_key_data *keysets, krb5_int16 n_keysets,
+                         int full_free)
 {
-    char                        *user=NULL;
-    int                         i=0, j=0, noofkeys=0;
-    krb5_key_data               *key_data=NULL, *tmp;
-    krb5_error_code             st=0;
+    int i;
+
+    if (keysets != NULL) {
```

Comment:
In situations like this, it's usually our preference to de-indent the contents of the if block by returning if keysets == NULL.